### ISSUE TITLE
[Encoder–decoder] Independently initialize cloned TransformerDecoder layers

### ISSUE THREAD

**Issue by gkondas** [mmcdermott/EveryQuery-conditional#18]
## Summary

`ConditionalQueryEncoderDecoderModel` builds one `torch.nn.TransformerDecoderLayer` and passes it to `torch.nn.TransformerDecoder`:

```python
decoder_layer = torch.nn.TransformerDecoderLayer(
    d_model=H,
    nhead=n_heads,
    dim_feedforward=decoder_ffn_mult * H,
    dropout=mlp_dropout,
    batch_first=True,
    norm_first=True,
)
self.decoder = torch.nn.TransformerDecoder(
    decoder_layer,
    num_layers=decoder_layers,
    norm=torch.nn.LayerNorm(H),
)
```

Source: [`conditional_model.py`](https://github.com/mmcdermott/EveryQuery-conditional/blob/claude/decoder-only-llamma-model-lm1vvk/src/every_query/model/conditional_model.py)

`TransformerDecoder` deep-copies that template layer. The copies have distinct `Parameter` objects/storage—they are **not weight-tied**—but every corresponding parameter starts with exactly the same value.

PyTorch explicitly warns about this behavior:

> All layers in the TransformerDecoder are initialized with the same parameters. It is recommended to manually initialize the layers after creating the TransformerDecoder instance.

Docs: https://docs.pytorch.org/docs/stable/generated/torch.nn.TransformerDecoder.html

## Why this may matter

The decoder layers are intended to learn different transformations, but identical starting values can delay layer specialization and alter optimization dynamics. The symmetry is not guaranteed to persist: layers operate at different depths, receive different inputs, and therefore generally get different gradients. Consequently, this is a plausible optimization concern, not by itself proof of a training failure.

This is also separate from the auxiliary-embedding initialization addressed by #17. It applies only to the encoder–decoder query decoder; it does not explain a decoder-only Llama run plateauing.

## Proposed change

After constructing `self.decoder`, independently initialize each cloned layer using a deliberate, documented policy.

One simple policy, similar to initialization used by PyTorch's reference Transformer, is to independently apply Xavier initialization to every matrix-valued parameter:

```python
for layer in self.decoder.layers:
    for parameter in layer.parameters():
        if parameter.dim() > 1:
            torch.nn.init.xavier_uniform_(parameter)
```

Biases and LayerNorm scale/offset can retain their conventional zero/one initialization. If preserving every native `TransformerDecoderLayer` submodule initializer exactly is preferred, an alternative is to construct the layers independently in a custom stack rather than cloning one initialized template. The chosen policy should be explicit because blanket Xavier initialization differs from the default initialization of some `Linear` modules.

## Tests

With `decoder_layers >= 2`:

- Assert corresponding matrix parameters in different decoder layers are not value-identical immediately after construction, e.g.:
  ```python
  assert not torch.equal(
      model.decoder.layers[0].self_attn.in_proj_weight,
      model.decoder.layers[1].self_attn.in_proj_weight,
  )
  ```
- Keep deterministic construction: resetting the global seed before constructing two models should still produce identical complete model state dicts.
- Optionally add a small overfit/smoke test to ensure the initialization change does not impair convergence.

## Compatibility

This changes initialization for new runs only. Existing checkpoints should continue to load unchanged.
